### PR TITLE
[core] Add emit support for all actor logic creators

### PR DESCRIPTION
**Pull request by davidkpiano**
This PR adds support for emitting events from all actor logic types:

**Promise actors**

```ts
const logic = fromPromise(async ({ emit }) => {
  // ...
  emit({
    type: 'emitted',
    msg: 'hello'
  });
  // ...
});
```

**Transition actors**

```ts
const logic = fromTransition((state, event, { emit }) => {
  // ...
  emit({
    type: 'emitted',
    msg: 'hello'
  });
  // ...
  return state;
}, {});
```

**Observable actors**

```ts
const logic = fromObservable(({ emit }) => {
  // ...

  emit({
    type: 'emitted',
    msg: 'hello'
  });

  // ...
});
```

**Callback actors**

```ts
const logic = fromCallback(({ emit }) => {
  // ...
  emit({
    type: 'emitted',
    msg: 'hello'
  });
  // ...
});
```

### CODE REVIEW

File: packages/core/src/actors/callback.ts [hunk 1]
```diff
@@ -171,7 +176,8 @@ export function fromCallback<
         receive: (listener) => {
           callbackState.receivers ??= new Set();
           callbackState.receivers.add(listener);
-        }
+        },
+        emit
```
Comment (Andarist) [inline packages/core/src/actors/callback.ts:180]: could you add a test that would successfully `emit` from a *rehydrated* callback actor?

Comment (davidkpiano) [reply packages/core/src/actors/callback.ts:180]: I will do this in a separate PR

Comment (Andarist) [reply packages/core/src/actors/callback.ts:180]: 1. create a machine with a callback actor
2. persist the root root actor
3. rehydrate a new instance of the root actor using the persisted snapshot
4. now verify that the *rehydrated* actor can correctly use `emit` after step 3

Comment (davidkpiano) [reply packages/core/src/actors/callback.ts:180]: 53c3c4dd58d29cf003959ffbd5ae279a01cd2ece

File: .changeset/smooth-crabs-dress.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+---
+'xstate': minor
+---
+
+All actor logic creators now support emitting events:
```
Comment (with-heart) [inline .changeset/smooth-crabs-dress.md:5]:
```suggestion
All actor logic creators now support [emitting events](https://stately.ai/docs/event-emitter):
```

**Review by Andarist — APPROVED** [review]

**Pull request merged**